doc/metadata-server/README.md

```diff
@@ -0,0 +1,327 @@
+
+```js
+signature = sign(hash(insId + CBOR(metadata)))
+```
+
+#### Template metadata
+
+```js
+extContractId = hash(CBOR(serialize(extContract)))
+
+extMetadata = {
+  template = {
+    contract: <extContract>,
+    metadata: {
+      shortDescription: <string>,
+      longDescription: <string>,
+      choiceDescriptions: [{ id: <string>, description: <string> }],
+      roleDescriptions: [{ id: <string>, description: <string> }],
+      valueDescriptions: [{ id: <string>, description: <string> }],
+
+      <other properties>
+    }
+  },
+  "params": [{ id: <string>, description: <string> }]
+  "signatures": <string | array>
+}
+```
+`signatures` and `params` are optional as well as `template.contract`. `template.metadata` should not be empty. We consider signature to be relevant to the transaction when:
+
+```js
+signature = sign(hash(extContractId + CBOR(extMetadata.template.metadata)))
+```
+
+and
+
+```js
+txContract == fillTemplate(extMetadata.template.contract, extMetadata.params)
+```
+
+#### Tx metadata schema
+
+```js
+contractMetadata = {"instance": <insMetadata>, "extended": <extMetadata>}
+
+txMetadata = {
+  "1564": {
+      "on": <contractMetadata>
+      "off": {
+        url: <string>
+        subject: <string>
+      }
+    }
+  }
+```
+
+- `subject` - optional value. By default `insId` should be used.
+- `url` - metadata server which provides possibly two entries:
+  - `/metadata/{subject}/property/instance`
+  - `/metadata/{subject}/property/extended`
+
+If both `on` and `off` parts are defined we treat `on` values as fallback ([`sequnenceNumber = 0`](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0026#sequence-numbers)) and prioritize responses from the server. We allow servers to respond with local redirects so entries can be lifted to subjects or even different REST resources i.e. `/metadata/{subject}/property/exteneded` could possibly redirect to `/extended/{extContractId}/`.
+
+## Implementation
+
+### Prototyping metadata server
+
+For an initial protyping phase of the metadata server we can use libraries from (`input-output-hk/offchain-metadata-tools`)[`https://github.com/input-output-hk/offchain-metadata-tools`]. This project provides a basic server and simple database layer (based on `postgresql`) for key-value service which implements [CIP-0026 - Cardano Off-Chain Metadata](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0026).
+In the case of Marketplace we want to probably expose only metadata subjects for the instances of contracts (like `/metadata/{ insId }/`) and expose templates and their metadata using other routing patterns (like `/extended/{ extContractId }/`) and use redirects for properties (`/metadata/{ insId }/property/extended`). Of course if we want to flatten both resources into the metadata subjects we can do this by using some encoding for inheritance on the `SQL` level but this comes with its own overhead and makes typing and schema of the API more complicated.
```

Comment:
I've been disappointed by the lack of Cardano's adoption of RDF (Resource Description Framework) metadata in general and Dublin Core Metadata in particular. Those are well thought out and extensible standards that align with decentralization. Should this document mention that semantic web metadata standards are available and are consistent with this proposal?